### PR TITLE
Tidy tokens endpoint

### PR DESCRIPTION
**Pull request by insacuri**
# Motivation and Context
The tokens endpoint sent hardcoded data that wasn't used anywhere.  This PR tidies it up whilst retaining the functionality for now.

# What has changed
The /tokens endpoint returns a 204 and no content, where it used to return a 200 with a json response that contained hardcoded values.

# How to test?
Look at the how to test part of https://github.com/ONSdigital/ras-frontstage/pull/459 for a detailed guide on how to test this

# Links
https://trello.com/c/XqLdUYwd/1032-tidy-up-tokens-endpoint-in-ras-rm-auth-service-3

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -77,6 +77,11 @@ An `HTTP 401 UNAUTHORIZED` status code is returned for unauthorized user credent
 
 
 ## verify account
+This endpoint came about from the old oauth2 auth service this service is replacing.  In that service, you hit the endpoint
```
Comment (necrophonic) [inline API.md]: :art: Markdown lint - should have a blank line after a heading

File: ras_rm_auth_service/resources/tokens.py [hunk 2]
```diff
@@ -25,6 +25,16 @@ def before_tokens_view():
...
+    """This endpoint looks weird because it has a history with the previous authentication service.
+    The previous one had a /tokens endpoint to provide oauth tokens for users who provided correct credentials.
+    In the changeover from the old auth service to this one, to make the changeover less risky, we kept this endpoint
+    with the same name, but instead of it returning a json payload with tokens, we decided a 204 would suffice as
+    the HTTP equivalent of a thumbs up that the credentials were correct.
+
+    Once the old service has been retired, this endpoint and this services API as a whole needs to be reviewed
+    and cleaned up.
+    """
+    logger.info("Verifying user credentials")
```
Comment (necrophonic) [inline ras_rm_auth_service/resources/tokens.py]: Is there any context we can add to this message? It's not really helpful on it's own out of context in the logs.

File: ras_rm_auth_service/resources/tokens.py [hunk 3]
```diff
@@ -48,6 +58,5 @@ def post_token():
             logger.debug(ex.description, username=payload.get('username'))
             return make_response(jsonify({"detail": ex.description}), 401)
 
-    return make_response(
-        jsonify({"id": 895725, "access_token": "NotImplementedInAuthService", "expires_in": 3600,
-                 "token_type": "Bearer", "scope": "", "refresh_token": "NotImplementedInAuthService"}), 201)
+    logger.info("User credentials correct")
```
Comment (necrophonic) [inline ras_rm_auth_service/resources/tokens.py]: Context?

File: API.md (hunk 1 above)
Comment (insacuri) [reply API.md]: Done

File: ras_rm_auth_service/resources/tokens.py [hunk 4]
```diff
@@ -33,20 +42,41 @@ def post_token():
...
                 401)
 
+        bound_logger.info("User found")
         try:
             user.authorise(payload.get('password'))
         except Unauthorized as ex:
+<<<<<<< HEAD
+            bound_logger.info("User is unauthorised", description=ex.description)
+=======
+>>>>>>> origin/master
```
Comment (necrophonic) [inline ras_rm_auth_service/resources/tokens.py]: You've commit merge artifacts

File: test/resources/test_tokens.py [hunk 5]
```diff
@@ -237,3 +233,13 @@ def test_post_tokens_empty_username_bad_request(self):
         # Then
         self.assertEqual(response.status_code, 400)
         self.assertEqual(response.get_json(), {"detail": "Missing 'username' or 'password'"})
+
+    @pytest.mark.parametrize('email, obfuscated_email', [
+        ('example@example.com', 'e*****e@e*********m'),
+        ('prefix@domain.co.uk', 'p****x@d*********k'),
+        ('first.name@place.gov.uk', 'f********e@p********k')
```
Comment (necrophonic) [inline test/resources/test_tokens.py]: Can you add a few more test examples with numbers and other characters? e.g:
```
me+addition@gmail.com
a.b.c.someone@example.com
john.smith123456@londinium.ac.co.uk
me!?@example.com
```

File: ras_rm_auth_service/resources/tokens.py (hunk 4 above)
Comment (insacuri) [reply ras_rm_auth_service/resources/tokens.py]: Thanks vscode!  I'll get that fixed ^^

File: test/resources/test_tokens.py (hunk 5 above)
Comment (insacuri) [reply test/resources/test_tokens.py]: Done

File: ras_rm_auth_service/resources/tokens.py (hunk 4 above)
Comment (insacuri) [reply ras_rm_auth_service/resources/tokens.py]: Done

**Pull request merged**